File: Junit - RSpec/JUnit/src/main/java/TareaConversion/StringCalculator.java

package TareaConversion;

//Conversion del ejemplo de Ruby(Rspec) a Java (JUnit)
public class StringCalculator {
    public StringCalculator(){}
    
    public int add(String input){
        return 0;
    }
    
    public int addTwo(String input){
        if(input.isEmpty()){
            return 0;
        }else{
            return Integer.parseInt(input);
        }
    }
    
    public int addThree(String input){
        if(input.isEmpty()){
            return 0;
        }else{
            String[] numbers = input.split(",");
            int sum = 0;
            for (int i=0;i<numbers.length;i++) {
                sum = sum + Integer.parseInt(numbers[i]);
            }
            return sum;
        }
    }
}

File: Junit - RSpec/RSpec/lib/string_calculator.rb
class StringCalculator
  
  def self.add(input)
    0    
  end

  def self.addTwo(input)
    if input.empty?
      0
    else
      input.to_i
    end
  end

  def self.addThree(input)
    if input.empty?
      0
    else
      numbers = input.split(",").map { |num| num.to_i }
      numbers.inject(0) { |sum, number| sum + number }
    end
  end
end